cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/TypeResolver.kt

```diff
@@ -30,190 +30,37 @@ import de.fraunhofer.aisec.cpg.graph.Component
 import de.fraunhofer.aisec.cpg.graph.Node
 import de.fraunhofer.aisec.cpg.graph.declarations.RecordDeclaration
 import de.fraunhofer.aisec.cpg.graph.types.*
-import de.fraunhofer.aisec.cpg.graph.types.HasSecondaryTypeEdge
 import de.fraunhofer.aisec.cpg.helpers.SubgraphWalker.IterativeGraphWalker
 
+/**
+ * The purpose of this [Pass] is to establish a relationship between [Type] nodes (more specifically
+ * [ObjectType]s) and their [RecordDeclaration].
+ */
```

Comment:
I'm wondering if we still need this pass or if we could just set/generate the `Type` to `RecordDeclaration` mapping when we parse the `RecordDeclaration`?

Reply:
Not quite sure, I would explore this in a separate PR if this is possible.